### DISCUSSION TITLE
Cut Count

### DISCUSSION THREAD

**Discussion by NathanChristie** [lairdubois/lairdubois-opencutlist-sketchup-extension#237] [Ideas]
Is there any way to identify and present the current amount of cuts required for parts?

So if I'm cutting a cabinet out of one sheet of plywood, perhaps there is a way to cut all parts in 20 cuts instead of 25. This could be added as an option under "Preordering of Parts" and possibly presented in the Cutting Diagram.

Thank your for all of your hard work on an amazing tool!

**Comment by mobilarte** [comment]
We do display the "Length of cuts" at the bottom of each panel, but not the nb of cuts.

Our version 2.0 algorithm computes between 48 and 768 different placements and then selects the "best" placement based on some criteria like compactness, nb of leftovers, nb of cuts, length of cuts, largest leftover area, .... I put "best" in quotes because optimality is difficult to define in this case. We had two must-criteria

1. the solution must be deterministic, no randomness is allowed (not in time, not in space). The algorithm must under any circumstances produce the same solution for an identical input.
2. the solution must look "good", we are trying to avoid solutions that are too convoluted.

These criteria eliminate many algorithms that may find close to optimal solutions in terms of minimizing space.  Considering that Bin Packing is an NP-hard problem, we are not guaranteed to find an optimal solution even if we were to concentrate on a single criterion.

Consider the following solution

![Clipboard10](https://user-images.githubusercontent.com/24858423/106336055-625be600-628e-11eb-8450-7d572b092fe6.jpg)

and compare it to this solution

![Clipboard11](https://user-images.githubusercontent.com/24858423/106336193-ae0e8f80-628e-11eb-8560-a9c781efea52.jpg)

Although the Length of cuts is larger for the second solution (by at least 5m - you could make it smaller with a few vertical cuts) and the leftover at the bottom is smaller, I think it is a better and easier solution to cut.

But there is really not guarantee that OpenCutList will find the absolute best solution. The purpose of the cutting diagram is to find an acceptable solution in a short time, but there is no obligation to follow it!

Version 2.0 will have less options to select from, because too many choices make us unhappy...

**Reply by NathanChristie** [reply]
Thank you for the very detailed explanation and images. These were very enlightening.
